data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/SerializedJson.java

```diff
@@ -25,4 +27,19 @@ static SerializedJson fromString(String jsonString) {
         Objects.requireNonNull(jsonString);
         return new SerializedJsonImpl(jsonString.getBytes(StandardCharsets.UTF_8));
     }
+
+    /**
+     * Creates a new {@link SerializedJson} from a JSON string and optional documentId and routingField.
+     *
+     * @param jsonString The serialized JSON string which forms this JSON data.
+     * @param docId Optional documment ID string
+     * @param routingField Optional routing field string
+     * @return A new {@link SerializedJson}.
+     */
+    static SerializedJson fromStringAndOptionals(String jsonString, String docId, String routingField) {
```

Comment:
I don't think we need to keep the old `fromString`. So this should probably replace it.